app/code/Magento/Catalog/view/frontend/templates/product/image_with_borders.phtml

```diff
@@ -27,25 +28,18 @@ $paddingBottom = $block->getRatio() * 100;
 $styles = <<<STYLE
 .product-image-container-{$block->getProductId()} {
     width: {$width}px;
+    height: auto;
+    aspect-ratio: {$width} / {$height};
 }
 .product-image-container-{$block->getProductId()} span.product-image-wrapper {
-    padding-bottom: {$paddingBottom}%;
+    height: 100%;
+    width: 100%;
 }
-STYLE;
-//In case a script was using "style" attributes of these elements
-$script = <<<SCRIPT
-prodImageContainers = document.querySelectorAll(".product-image-container-{$block->getProductId()}");
-for (var i = 0; i < prodImageContainers.length; i++) {
-    prodImageContainers[i].style.width = "{$width}px";
-}
-prodImageContainersWrappers = document.querySelectorAll(
-    ".product-image-container-{$block->getProductId()}  span.product-image-wrapper"
-);
-for (var i = 0; i < prodImageContainersWrappers.length; i++) {
-    prodImageContainersWrappers[i].style.paddingBottom = "{$paddingBottom}%";
+@supports not (aspect-ratio: auto) { 
+    .product-image-container-{$block->getProductId()} span.product-image-wrapper {
+        padding-bottom: {$paddingBottom}%;
+    }
```

Comment:
well support aspect ratio at this moment so i think we don't have to hacky approach with padding

Reply:
we even not need $block->getRatio()

Reply:
Nice! From preliminary testing it seems this could be a winner. Will post more in-depth in a normal comment below.